js/ui/scheduler/workspaces/ui.scheduler.work_space.js

```diff
@@ -364,10 +374,21 @@ const SchedulerWorkSpace = Widget.inherit({
 
     _getGroupIndexByCell: function($cell) {
         const cellsInRow = this._getCellCount();
-        const currentCellIndex = $cell.index() + 1;
-        const groupIndex = Math.ceil(currentCellIndex / cellsInRow);
+        const columnIndex = $cell.index();
+        const groupOrientation = this.option('groupOrientation');
+        const groupCount = this._getGroupCount();
 
-        return groupIndex;
+        if(this.isGroupedByDate()) {
+            return columnIndex % groupCount;
+        }
+        if(groupOrientation === 'vertical') {
```

Comment:
Can we move this code to VerticalGroupedStrategy, separate logic between available strategies?